airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalAttemptExecution.java

```diff
@@ -44,31 +52,42 @@
  * outputs are passed to the selected worker. It also makes sures that the outputs of the worker are
  * persisted to the db.
  */
-public class TemporalAttemptExecution<T> implements CheckedSupplier<T, TemporalJobException> {
+public class TemporalAttemptExecution<INPUT, T> implements CheckedSupplier<T, TemporalJobException> {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(TemporalAttemptExecution.class);
+  private static final Duration HEARTBEAT_INTERVAL = Duration.ofSeconds(10);
 
   private final Path jobRoot;
-  private final CheckedFunction<Path, T, Exception> execution;
+  private final CheckedFunction<Path, Worker<INPUT, T>, Exception> workerSupplier;
+  private final Supplier<INPUT> inputSupplier;
   private final long jobId;
   private final BiConsumer<Path, Long> mdcSetter;
   private final CheckedConsumer<Path, IOException> jobRootDirCreator;
+  private final CancellationHandler cancellationHandler;
 
   @VisibleForTesting
-  TemporalAttemptExecution(Path workspaceRoot, JobRunConfig jobRunConfig, CheckedFunction<Path, T, Exception> execution) {
-    this(workspaceRoot, jobRunConfig, execution, WorkerUtils::setJobMdc, Files::createDirectories);
+  TemporalAttemptExecution(Path workspaceRoot,
+                           JobRunConfig jobRunConfig,
+                           CheckedFunction<Path, Worker<INPUT, T>, Exception> workerSupplier,
```

Comment:
i think the objective here is to catch that _any_ exception that is thrown (checked or unchecked). It could be that it would be more coherent to remove Exception from the method signature if it not being used and instead  catch all runtime exceptions and convert them to temporal exceptions.